add JSC section for ISC'22 tutorial (#99)
easybuilders/easybuild-tutorial: docs/2022-isc22/integration_jsc.md

performance analysis of parallel programs, visualization, computational steering and
federated data services.

In cooperation with hardware and software vendors like IBM, ATOS, Intel, NVIDIA and ParTec,
JSC meets the challenges that arise from the development of exaflop systems - the
computers of the next supercomputer generation. As a member of the German Gauss Centre
for Supercomputing, JSC has also coordinated the construction
of the European research infrastructure "PRACE - Partnership for Advanced Computing in
Europe" since 2008.

### Resources

JSC currently manages 3 primary systems (in addition to a
number of other development clusters):

[JUWELS](https://apps.fz-juelich.de/jsc/hps/juwels/configuration.html)
is a milestone on the road to a new generation of ultra-flexible modular supercomputers
targeting a broader range of tasks. It currently has 10.6 (CPU) + 1.7 (GPU) Petaflop per
second peak performance. In the coming months, JUWELS will become the first
supercomputer equipped with NVIDIA A100 GPU, when a new GPU module providing an
additional 58 Petaflops will be installed. This module will make JUWELS the most

> powerful supercomputer in Europe.

Reviewer: Pretty sure this is outdated...

> second peak performance. In the coming months, JUWELS will become the first
> supercomputer equipped with NVIDIA A100 GPU, when a new GPU module providing an
> additional 58 Petaflops will be installed. This module will make JUWELS the most
> powerful supercomputer in Europe.

Reviewer: ```suggestion
second peak performance on the cluster module and an NVIDIA GPU booster module
providing an additional 73 Petaflops. This module makes JUWELS the most
powerful supercomputer in Europe.
```

and batch jobs. The cloud partition will enable co-location of (web) services with these
resources to enable new workflows and support community platforms.

## Usage of EasyBuild within JSC

As a large site with multiple systems and diverse requirements, JSC takes advantage of
how easily EasyBuild
can be extensively configured according to site policies, ranging from the software
installation prefix to
all aspects of the module naming scheme being used for the modules being generated.

JSC maintains a [public repository of the customisations and development environment
for EasyBuild](https://github.com/easybuilders/JSC) that we use in our production
environment. Below we highlight some particular cases of these customisations.

### Custom toolchains

As of June 2022,
there are a total of 16 unique toolchain definitions in use at JSC, which reflect
multiple
combinations of compilers (`GCCcore`, `GCC`, `Intel` and `NVHPC`),
MPI runtimes (`ParaStationMPI`, `OpenMPI` and `IntelMPI`)
and mathematical libraries (`MKL`).

Given the proliferation of toolchains required at our site, JSC has put a lot of effort

> into increasing the capabilities of the `--try-toolchain` option and has recently
> introduced the `--try-update-deps` experimental option to more easily adopt upstream changes and adapt them to our

Reviewer: ```suggestion
into increasing the capabilities of the `--try-toolchain` option and
introduced the `--try-update-deps` experimental option to more easily adopt upstream changes and adapt them to our
```

etc.). Arising from this and the
custom toolchains that we use, we maintain our own reference easyconfig repository (our
*Golden* repository).

We are actively trying to minimise the differences between the two (see our
[usage of hooks](#usage-of-hooks) below) as we recognise that this
introduces an additional maintenance burden for us, and inhibits our ability to
easily contribute back our easyconfigs to EasyBuild.

### Hiding dependencies

While we provide an extensive set of software, we try to minimize the packages exposed
to the users by hiding a large set of dependencies which users
are unlikely to require directly (via the `hide-deps` configuration setting). There are currently
over 200 such hidden dependencies.  

While hidden dependencies are not visible in the `module` view by default, users can
expose them by the use of the `--show-hidden` argument in `Lmod`:
```shell
module --show-hidden avail
``` 

### Usage of hooks

The relatively new [***hooks*** feature](https://docs.easybuild.io/en/latest/Hooks.html)
of EasyBuild provides JSC with an opportunity to
track upstream developments more closely.

> We are currently integrating a new hook that provides a lot of useful functionality:

Reviewer: ```suggestion
We have integrated, and continue to develop, a hook that provides a lot of useful functionality:
```
